zerver/models.py

```diff
@@ -214,8 +215,9 @@ def remote_user_to_email(remote_user):
 
 class RealmEmoji(models.Model):
     realm = models.ForeignKey(Realm)
-    name = models.TextField()
-    img_url = models.TextField()
+    name = models.TextField(validators=[MinLengthValidator(1),
+                                        RegexValidator(regex=r'^[0-9a-zA-Z_\-\.]*(?<![.\-_])$')])
```

Comment:
Maybe add a comment explaining the second part of this regexp is enforcing that the name not end with a punctuation character?